pagerduty/resource_pagerduty_team_membership.go

```diff
@@ -38,11 +44,12 @@ func resourcePagerDutyTeamMembershipCreate(d *schema.ResourceData, meta interfac
 
 	userID := d.Get("user_id").(string)
 	teamID := d.Get("team_id").(string)
+	role := d.Get("role").(string)
 
-	log.Printf("[DEBUG] Adding user: %s to team: %s", userID, teamID)
+	log.Printf("[DEBUG] Adding user: %s to team: %s with role: %s", userID, teamID, role)
 
 	retryErr := resource.Retry(2*time.Minute, func() *resource.RetryError {
-		if _, err := client.Teams.AddUser(teamID, userID); err != nil {
+		if _, err := client.Teams.AddUserWithRole(teamID, userID, role); err != nil {
```

Comment:
What if the role is empty? Shouldn't `client.Teams.AddUser(teamID, userID)` be used instead?

Reply:
I defined a default value for `role` to be the same as the current API behavior, calling AddUser API without `role`.
https://github.com/terraform-providers/terraform-provider-pagerduty/pull/151/commits/a560e09da562118a90fe34f668cfc33a9cc2918a

With validation implemented at https://github.com/terraform-providers/terraform-provider-pagerduty/commit/8521226ac6405fd3bf2c69c48aeb325b3cdb4d23, this resource guarantees a valid value is set in `role` at every time. So I think it's okay to use `AddUserWithRole` at every time too.